pkg/controller/cloud/node_controller.go

```diff
@@ -239,9 +241,38 @@ func (cnc *CloudNodeController) MonitorNode() {
 		// from the cloud provider. If node cannot be found in cloudprovider, then delete the node immediately
 		if currentReadyCondition != nil {
 			if currentReadyCondition.Status != v1.ConditionTrue {
+				// we need to check this first to get taint working in similar in all cloudproviders
+				// current problem is that shutdown nodes are not working in similar way ie. all cloudproviders
+				// does not delete node from kubernetes cluster when instance it is shutdown see issue #46442
+				exists, err := instances.InstanceShutdownByProviderID(node.Spec.ProviderID)
+				if err != nil {
+					glog.Errorf("Error getting data for node %s from cloud: %v", node.Name, err)
+					continue
+				}
+
+				if exists {
+					// if node is shutdown add shutdown taint
+					cloudTaint := getTaint(node.Spec.Taints, algorithm.TaintNodeShutdown)
```

Comment:
You should use [`AddOrUpdateTaintOnNode`](https://github.com/kubernetes/kubernetes/blob/d6325933e1b5ac8a2c0a5a65ab8bd690015f4697/pkg/controller/controller_utils.go#L893) and remove most of this logic including the patch method and undo the changes to `getCloudTaint`.